plotly/tests/test_optional/test_offline/test_offline.py

```diff
@@ -22,3 +40,61 @@ def test_iplot_works_after_you_call_init_notebook_mode(self):
         plotly.tools._ipython_imported = True
         plotly.offline.init_notebook_mode()
         plotly.offline.iplot([{}])
+
+    def test_iplot_mpl_works_after_you_call_init_notebook_mode(self):
+        plotly.tools._ipython_imported = True
+        plotly.offline.init_notebook_mode()
+        plotly.offline.iplot_mpl(fig)
+
+
+class PlotlyOfflineMPLTestCase(TestCase):
+    def setUp(self):
+        pass
+
+    def _read_html(self, file_url):
+        """ Read and return the HTML contents from a file_url in the
+        form e.g. file:///Users/chriddyp/Repos/plotly.py/plotly-temp.html
+        """
+        with open(file_url.replace('file://', '').replace(' ', '')) as f:
+            return f.read()
+
+    def test_default_mpl_plot_generates_expected_html(self):
+        data_json = ('[{"name": "_line0", "yaxis": "y1", "marker": {"color":' +
+                     ' "#0000FF", "opacity": 1, "line": {"color": "#000000",' +
+                     ' "width": 0.5}, "symbol": "dot", "size": 6.0}, "mode":' +
+                     ' "markers", "xaxis": "x1", "y": [100.0, 200.0, 300.0],' +
+                     ' "x": [10.0, 20.0, 30.0], "type": "scatter"}]')
+        layout_json = ('{"autosize": false, "width": 640, "showlegend": ' +
+                       'false, "xaxis1": {"tickfont": {"size": 12.0}, ' +
+                       '"domain": [0.0, 1.0], "ticks": "inside", "showgrid":' +
+                       ' false, "range": [10.0, 30.0], "mirror": "ticks", ' +
+                       '"zeroline": false, "showline": true, "nticks": 5, ' +
+                       '"type": "linear", "anchor": "y1", "side": "bottom"},' +
+                       ' "height": 480, "yaxis1": {"tickfont": ' +
+                       '{"size": 12.0}, "domain": [0.0, 1.0], "ticks": ' +
+                       '"inside", "showgrid": false, "range": [100.0, 300.0]' +
+                       ', "mirror": "ticks", "zeroline": false, "showline": ' +
+                       'true, "nticks": 5, "type": "linear", "anchor": "x1",' +
+                       ' "side": "left"}, "hovermode": "closest", "margin":' +
+                       ' {"b": 47, "r": 63, "pad": 0, "t": 47, "l": 80}}')
```

Comment:
At some point we should create a fixture file that has a largish/complicated plot in it which we can use over and over for testing... You probably don't need to go there now though :)